joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -524,6 +540,26 @@ CallbackReturn JointTrajectoryController::on_configure(const rclcpp_lifecycle::S
     return CallbackReturn::FAILURE;
   }
 
+  if (has_velocity_command_interface_ && use_closed_loop_pid_adapter)
+  {
+    pids_.resize(joint_names_.size());
+    velocity_ff_.resize(joint_names_.size());
```

Comment:
There should be `num_joints` define (or similar variable).

```suggestion
    pids_.resize(num_joints);
    velocity_ff_.resize(num_joints);
```

Reply:
ros-controls/ros2_controllers@68d838d